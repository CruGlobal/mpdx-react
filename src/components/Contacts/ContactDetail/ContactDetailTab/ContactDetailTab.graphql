query ContactDetailTab($accountListId: ID!, $contactId: ID!) {
  contact(accountListId: $accountListId, id: $contactId) {
    id
    name
    primaryPerson {
      id
    }
    primaryAddress {
      street
      city
      state
      postalCode
      country
    }
    tagList
    envelopeGreeting
    people {
      nodes {
        id
        firstName
        lastName
        primaryPhoneNumber {
          number
        }
        primaryEmailAddress {
          email
        }
        anniversaryDay
        anniversaryMonth
        anniversaryYear
        birthdayDay
        birthdayMonth
        birthdayYear
      }
    }
    preferredContactMethod
    timezone
    user {
      firstName
      lastName
    }
    website
  }
}
