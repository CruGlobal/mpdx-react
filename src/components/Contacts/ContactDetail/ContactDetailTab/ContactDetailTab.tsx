import { Box, Button, makeStyles, styled } from '@material-ui/core';
import { Create } from '@material-ui/icons';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { ContactDetailQuery } from '../ContactDetail.generated';

const ContactDetailSectionContainer = styled(Box)(({ theme }) => ({
  padding: theme.spacing(0),
  paddingTop: '4px',
  margin: 0,
  alignContent: 'start',
}));

const ContactDetailHeadingContainer = styled(Box)(() => ({
  display: 'flex',
  flexDirection: 'row',
  alignContent: 'center',
  alignItems: 'center',
}));

const ContactDetailHeadingIcon = styled(Create)(({ theme }) => ({
  width: '18px',
  height: '18px',
  margin: theme.spacing(0),
  color: '#9C9FA1',
}));

const ContactDeleteButton = styled(Button)(({ theme }) => ({
  display: 'block',
  padding: theme.spacing(2),
  margin: theme.spacing(5),
  marginLeft: 'auto',
  marginRight: 'auto',
}));

const useStyles = makeStyles(() => ({
  contactDetailHeadingText: {
    flexGrow: 5,
  },
}));

interface ContactDetailTabProps {
  contact: ContactDetailQuery;
}

export const ContactDetailTab: React.FC<ContactDetailTabProps> = ({
  contact,
}) => {
  const classes = useStyles();
  const calledContact = contact.contact;
  const { t } = useTranslation();

  return (
    <Box style={{ width: '100%', paddingRight: '5%', paddingLeft: '5%' }}>
      {
        // Tag Section
      }
      <ContactDetailHeadingContainer></ContactDetailHeadingContainer>
      <hr />
      {
        // People Section
      }
      <ContactDetailSectionContainer>
        <ContactDetailHeadingContainer>
          <h6 className={classes.contactDetailHeadingText}>
            {calledContact.name}
          </h6>
          <ContactDetailHeadingIcon />
        </ContactDetailHeadingContainer>
      </ContactDetailSectionContainer>
      <hr />
      {
        // Mailing Section
      }
      <ContactDetailSectionContainer>
        <ContactDetailHeadingContainer>
          <h6 className={classes.contactDetailHeadingText}>{t('Mailing')}</h6>
          <ContactDetailHeadingIcon />
        </ContactDetailHeadingContainer>
      </ContactDetailSectionContainer>
      <hr />
      {
        // other Section
      }
      <ContactDetailSectionContainer>
        <ContactDetailHeadingContainer>
          <h6 className={classes.contactDetailHeadingText}>{t('Other')}</h6>
          <ContactDetailHeadingIcon />
        </ContactDetailHeadingContainer>
      </ContactDetailSectionContainer>
      <hr />
      <ContactDeleteButton variant="outlined" color="default">
        {t('DELETE CONTACT').toLocaleUpperCase()}
      </ContactDeleteButton>
    </Box>
  );
};
