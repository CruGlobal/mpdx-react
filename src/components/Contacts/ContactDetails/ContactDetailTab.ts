export enum ContactDetailTabEnum {
  Tasks = 'Tasks',
  Donations = 'Donations',
  Referrals = 'Referrals',
  ContactDetails = 'ContactDetails',
  Notes = 'Notes',
}
