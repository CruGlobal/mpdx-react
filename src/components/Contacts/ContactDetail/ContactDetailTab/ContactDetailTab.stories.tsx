import React from 'react';
import { Story } from '@storybook/react';
import { gqlMock } from '../../../../../__tests__/util/graphqlMocking';
import {
  ContactDetailDocument,
  ContactDetailQuery,
} from '../ContactDetail.generated';
import { ContactDetailTab } from './ContactDetailTab';

export default {
  title: 'Contacts/Tab/ContactDetailTab',
  component: ContactDetailTab,
};

export const Default: Story = () => {
  const contact = gqlMock<ContactDetailQuery>(ContactDetailDocument, {
    mocks: {},
  });
  return <ContactDetailTab contact={contact} />;
};
