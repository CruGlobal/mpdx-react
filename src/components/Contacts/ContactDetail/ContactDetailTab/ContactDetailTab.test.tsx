import { renderHook } from '@testing-library/react-hooks';
import { GqlMockedProvider } from '../../../../../__tests__/util/graphqlMocking';
import { useContactDetailTabQuery } from './ContactDetailTab.generated';

describe('ContactDetailTab', () => {
  it('test query', async () => {
    const { result, waitForNextUpdate } = renderHook(
      () =>
        useContactDetailTabQuery({
          variables: {
            accountListId: 'accountList-id',
            contactId: 'contact-id',
          },
        }),
      { wrapper: GqlMockedProvider },
    );
    await waitForNextUpdate();
    expect(result.current.variables).toMatchInlineSnapshot(`
      Object {
        "accountListId": "accountList-id",
        "contactId": "contact-id",
      }
    `);
    expect(result.current.data.contact.name).toMatchInlineSnapshot(
      `"Pyramid Car Rocket"`,
    );
  });
});
