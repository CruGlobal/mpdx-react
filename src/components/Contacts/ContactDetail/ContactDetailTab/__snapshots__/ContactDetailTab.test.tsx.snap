// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetailTab test query 2`] = `
Object {
  "__typename": "Contact",
  "envelopeGreeting": "Man Tunnel",
  "id": "2211577",
  "name": "Pyramid Car Rocket",
  "people": Object {
    "__typename": "PersonConnection",
    "nodes": Array [
      Object {
        "__typename": "Person",
        "anniversaryDay": 73,
        "anniversaryMonth": 31,
        "anniversaryYear": 38,
        "birthdayDay": 70,
        "birthdayMonth": 23,
        "birthdayYear": 95,
        "firstName": "Chisel",
        "id": "3965933",
        "lastName": "Compact Disc Boss",
        "primaryEmailAddress": Object {
          "__typename": "EmailAddress",
          "email": "Salt Tunnel Pyramid",
        },
        "primaryPhoneNumber": Object {
          "__typename": "PhoneNumber",
          "number": "Bomb Solid",
        },
      },
      Object {
        "__typename": "Person",
        "anniversaryDay": 46,
        "anniversaryMonth": 54,
        "anniversaryYear": 22,
        "birthdayDay": 84,
        "birthdayMonth": 64,
        "birthdayYear": 98,
        "firstName": "Tiger Rifle",
        "id": "4140142",
        "lastName": "Grapes",
        "primaryEmailAddress": Object {
          "__typename": "EmailAddress",
          "email": "Ice Earth Prison",
        },
        "primaryPhoneNumber": Object {
          "__typename": "PhoneNumber",
          "number": "Apple",
        },
      },
      Object {
        "__typename": "Person",
        "anniversaryDay": 68,
        "anniversaryMonth": 69,
        "anniversaryYear": 50,
        "birthdayDay": 3,
        "birthdayMonth": 71,
        "birthdayYear": 74,
        "firstName": "Eraser Comet Passport",
        "id": "7056502",
        "lastName": "Tennis racquet",
        "primaryEmailAddress": Object {
          "__typename": "EmailAddress",
          "email": "Plane Library",
        },
        "primaryPhoneNumber": Object {
          "__typename": "PhoneNumber",
          "number": "Snail Feather",
        },
      },
      Object {
        "__typename": "Person",
        "anniversaryDay": 81,
        "anniversaryMonth": 32,
        "anniversaryYear": 10,
        "birthdayDay": 16,
        "birthdayMonth": 10,
        "birthdayYear": 26,
        "firstName": "Treadmill Chocolates",
        "id": "1805674",
        "lastName": "Surveyor",
        "primaryEmailAddress": Object {
          "__typename": "EmailAddress",
          "email": "Slave Adult Staircase",
        },
        "primaryPhoneNumber": Object {
          "__typename": "PhoneNumber",
          "number": "Pebble Rope",
        },
      },
    ],
  },
  "preferredContactMethod": "FACEBOOK",
  "primaryAddress": Object {
    "__typename": "Address",
    "city": "Chair",
    "country": "Coffee Drill Cup",
    "postalCode": "Printer Coffee-shop",
    "state": "Diamond Feather Sex",
    "street": "Printer Coffee-shop",
  },
  "primaryPerson": Object {
    "__typename": "Person",
    "id": "9339842",
  },
  "tagList": Array [
    "Junk Bomb Leather jacket",
  ],
  "timezone": "Satellite Eyes God",
  "user": Object {
    "__typename": "UserScopedToAccountList",
    "firstName": "Baby Milk",
    "lastName": "Stomach",
  },
  "website": "Water Plane Chess Board",
}
`;
